### PR TITLE
[FIX] web_editor: convert link only a space after it

### PR DESCRIPTION
**Pull request by Goaman**
Before this commit, when adding a space anywhere in the a text node,
a plain text link gets transformed into a proper link.

It might be an undesired behavior as we could want to manually remove
the link but it will be re-transformed when completing a paragraph.

task-2795019

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 1]
```diff
@@ -2239,7 +2240,9 @@ export class OdooEditor extends EventTarget {
                     ev.data &&
                     ev.data.includes(' ') &&
                     selection &&
-                    selection.anchorNode
+                    selection.anchorNode &&
+                    selection.anchorNode.nodeType === Node.TEXT_NODE &&
+                    selection.anchorNode.textContent.slice(0, selection.anchorOffset).trim().split(' ').pop().match(URL_REGEX_WITH_INFOS)
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:2245]: Ca me semble fort ad-hoc comme check ici... Genre quid si c'est en plusieurs text nodes ? Ca marchera pas... Y'a pas moyen de faire un check un peu plus robuste que ça ?

Comment (Goaman) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js]: updated

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 2]
```diff
@@ -2237,13 +2238,32 @@ export class OdooEditor extends EventTarget {
                 // Check for url after user insert a space so we won't transform an incomplete url.
                 if (
                     ev.data &&
-                    ev.data.includes(' ') &&
+                    ev.data === ' ' &&
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:2241]: Will it work in mobile with composition ? Maybe it should check if it ends with a space ?

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 3]
```diff
@@ -2237,13 +2238,32 @@ export class OdooEditor extends EventTarget {
...
                         this.commandBar._currentOpenOptions.closeOnSpace !== true)
                 ) {
-                    this._convertUrlInElement(closestElement(selection.anchorNode));
+                    const text = selection.anchorNode.textContent.slice(0, selection.anchorOffset);
+                    const textNodeSplitted = text.split(/\s/);
+
+                    // Remove added space
+                    textNodeSplitted.pop();
+                    const potentialUrl = textNodeSplitted.pop();
+                    const lastWordMatch = new RegExp(URL_REGEX_WITH_INFOS).exec(potentialUrl);
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:2255]: This is different than all the places where `URL_REGEX_WITH_INFOS` is checked against.

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 4]
```diff
@@ -2237,13 +2238,32 @@ export class OdooEditor extends EventTarget {
...
+
+                    if (lastWordMatch) {
+                        const matches = getUrlsInfosInString(selection.anchorNode.textContent.slice(0, selection.anchorOffset));
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:2258]: It is not clear to me why you need to re-slice considering you already did it earlier.

**Comment by sebgeelen** [comment]
Where is the Unit test that check what was not working before your fix ?

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js (hunk 2 above)
Comment (dmo-odoo) [reply addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:2241]: Wait so it does not convert urls in mobile ? That's sad.

**Comment by Goaman** [comment]
> Where is the Unit test that check what was not working before your fix ?

I've added tests.

File: addons/web_editor/static/lib/odoo-editor/test/utils.js [hunk 5]
```diff
@@ -478,4 +481,81 @@ export function redo(editor) {
...
+    );
+    const EventClass = getEventConstructor(el.ownerDocument.defaultView, eventName);
+    if (EventClass.name === 'ClipboardEvent' && !('clipboardData' in options)) {
+        throw new Error('ClipboardEvent must have clipboardData in options');
+    }
+    const ev = new EventClass(eventName, options);
+
+    currentElement.dispatchEvent(ev);
+    return ev;
+}
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/test/utils.js:559]: I don't understand why all of this is needed ?

Comment (Goaman) [reply addons/web_editor/static/lib/odoo-editor/test/utils.js:559]: As talked in private, to have the function available in utils for future use. 
A refactor in master simplify current tests that could use this function.

**Comment by dmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**